src/vivarium_census_prl_synth_pop/components/household.py

```diff
@@ -51,47 +57,100 @@ def setup(self, builder: Builder):
             requires_columns=["household_id"],
         )
 
-    ########################
-    # Event-driven methods #
-    ########################
-
-    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+    def sample_initial_households(self, builder: Builder) -> pd.DataFrame:
```

Comment:
I'm curious: did you consider using the `set_households` method within `population.py` to keep this logic there? IMHO, it fits better there and at population initialization time than at setup time.

I'm pretty sure I can make household immigration work with this approach, but I haven't fully thought it through. Certainly in that case, both sampling of households and initialization of simulants will need to happen in one atomic action.

Reply:
Note for posterity:

* One end goal of these refactors is to avoid having to keep the ACS households data file in memory in the `Population` component for the whole duration of the sim.
* We'd prefer to only have households updates in "household" components and keep them out of `Population`. (Although `PersonMigration` and (future) `Immigration` components aren't explicitly household-focused, we consider them "household" components.)
* The analogous approach to this for household immigration will be to `set_households` with the new households being created before calling the simulant creator function, keeping the two updates decoupled while doing them as an atomic action (since simulant creation happens immediately).